### ISSUE TITLE
For templated regions, L003 complains that indentation is wrong, even though the raw (pre-template expansion) code looks correctly indented

### ISSUE THREAD

**Issue by pempey** [sqlfluff/sqlfluff#2270]
## Expected Behaviour

The indentation after a templated area would behave in a way that is visually similar to other lines.

## Observed Behaviour

When a column in a query is templated and then aliased with a single space after the templated area an L003 error is raised unless there is a full indentation after the templated area.

## Steps to Reproduce

Using the dbt templator and the included macro lint the following sql:

```sql
{% macro get_keyed_nulls(columns) %}

  COALESCE({{columns}}, MD5(-1)) 

{% endmacro %}

SELECT
    table_id,
    {{ get_keyed_nulls('other_id') }} AS other_id
FROM my_table
```

## Dialect

Snowflake

## Version
Include the output of `sqlfluff --version` along with your Python version.
If you are using dbt then please additionally include the dbt version.

```console
==== sqlfluff ====
sqlfluff:                0.9.0 python:                  3.8.6
implementation:        cpython dialect:             snowflake
verbosity:                   1 templater:                 dbt
dbt:                   =0.19.1
```

## Configuration
```
[sqlfluff]
dialect = snowflake
templater = dbt

[sqlfluff:templater:dbt]
project_dir = ./
```

**Comment by barrywhart** [comment]
Sample linter output from this query:
```
(sqlfluff-3.10.1) ➜  sqlfluff git:(main) ✗ sqlfluff lint test.sql 
== [test.sql] FAIL                                                                                                                                                                                      
L:   9 | P:  39 | L003 | Indentation not consistent with line #8
All Finished 📜 🎉!
```

IIUC, the concern is that SQLFluff says the indentation is wrong, but it must (?) be looking at line 3 indentation strangely, perhaps by combining the 2-space indent on line 3 (`COALESCE()`...) with the 4-space indent of line 9.

Intuitively, L003 here should look at the raw code (prior to template expansion) and check that indentation, not the post-template expansion indentation.

**Comment by barrywhart** [comment]
I think this issue can be fixed using a similar technique as the fix for #2046, i.e. translating the templated spaces back to raw code when computing the existing indentation. The coding will probably be trickier, though, because L003 (the rule involved in _this_ issue is pretty large and complex) compared to L009 (the rule in the other issue).

**Comment by barrywhart** [comment]
Simple test case that produces the same issue, works with the Jinja templater, and requires no `.sqlfluff` configuration file:
```
SELECT
    c1,
    {{ " c2" }}
FROM my_table
```

**Comment by barrywhart** [comment]
This code computes the leading spaces as well, and seems more robust, since it uses the slicing functions directly, rather than trusting that `.lstrip()` will stop in the correct place.

```
                    from sqlfluff.core.rules.functional import Segments
                    indent_raw_slices = Segments(*this_line["indent_buffer"], templated_file=context.templated_file).raw_slices
                    if "templated" in {slice_.slice_type for slice_ in indent_raw_slices}:
                        literal_indent = []
                        seg: BaseSegment
                        for seg in this_line["indent_buffer"]:
                            for pos in range(seg.pos_marker.source_slice.start, seg.pos_marker.source_slice.stop):
                                current_slice = slice(pos, pos + 1)
                                if context.templated_file.is_source_slice_literal(current_slice):
                                    literal_indent.append((current_slice, context.templated_file.source_str[current_slice]))
                                else:
                                    break
```

Running this on the example correctly extracts the first 4 spaces, then stops:
```
> /Users/bhart/dev/sqlfluff/src/sqlfluff/rules/L003.py(594)_process_current_line()
-> return LintResult(
(Pdb) pp literal_indent
[(slice(15, 16, None), ' '),
 (slice(16, 17, None), ' '),
 (slice(17, 18, None), ' '),
 (slice(18, 19, None), ' ')]
```

**Comment by pempey** [comment]
So I checked on this fix where I originally identified the issue and it has not been fixed.  My issue was related to the space after the templated region not being consistent.   The original steps to reproduce still product the same error.

**Comment by barrywhart** [comment]
Oh -- interesting! There were really two issues here. When I created a simpler test case:
```
SELECT
    c1,
    {{ "      c2" }}
FROM my_table
```

that produced a similar issue, which I fixed. But as you say, there's a second issue occurring between the templating and the `AS other_id` part. I'll reopen this and look into the second issue. Sorry about that!

**Comment by barrywhart** [comment]
This test case tests for the actual issue (note the added `\n` before `AS other_id`).
```
SELECT
    c1,
    {{ "      c2\n" }} AS other_id
FROM my_table
```
